File: ios/TicTacToe/Game/TicTacToeMove.swift
import GameplayKit

/// A move: place the active player's mark at `cellIndex`. Mirrors GameplayKit's
/// `GKGameModelUpdate`.
public final class TicTacToeMove: NSObject, GKGameModelUpdate {
    public let cellIndex: Int
    public var value: Int = 0

    public init(cellIndex: Int) {
        self.cellIndex = cellIndex
    }
}

File: android/app/src/main/kotlin/jp/co/bitz/tictactoe/game/TicTacToeMove.kt
package jp.co.bitz.tictactoe.game

import jp.co.bitz.gameplaykit.GKGameModelUpdate

/**
 * A move: place the active player's mark at [cellIndex]. Mirrors GameplayKit's
 * `GKGameModelUpdate`.
 */
public class TicTacToeMove(
    public val cellIndex: Int,
) : GKGameModelUpdate {
    override var value: Int = 0
}
